x-pack/plugins/security_solution/cypress/e2e/detection_rules/rule_fiters.cy.ts

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { cleanKibana, resetRulesTableState, deleteAlertsAndRules } from '../../tasks/common';
+import { login, visitWithoutDateRange } from '../../tasks/login';
+import { esArchiverResetKibana } from '../../tasks/es_archiver';
+import { findRuleRowInTable } from '../../tasks/rule_snoozing';
+import { expectRulesWithExecutionStatus, filterByExecutionStatus } from '../../tasks/rule_filters';
+
+import { SECURITY_DETECTIONS_RULES_URL } from '../../urls/navigation';
+
+import {
+  RULES_MANAGEMENT_TABLE,
+  RULE_EXECUTION_STATUS,
+} from '../../screens/alerts_detection_rules';
+
+import { waitForRulesTableToBeLoaded } from '../../tasks/alerts_detection_rules';
+
+import { createRule, waitForRulesToFinishExecution } from '../../tasks/api_calls/rules';
+import { deleteIndex, createIndex, indexDocument } from '../../tasks/api_calls/elasticsearch';
+
+import { getNewRule } from '../../objects/rule';
+
+describe('Rule management filters', () => {
+  before(() => {
+    cleanKibana();
+  });
+
+  beforeEach(() => {
+    login();
+    // Make sure persisted rules table state is cleared
+    resetRulesTableState();
+    deleteAlertsAndRules();
+    esArchiverResetKibana();
+
+    deleteIndex('test_index');
+
+    createIndex('test_index', {
+      '@timestamp': {
+        type: 'date',
+      },
+    });
+
+    indexDocument('test_index', {});
+
+    createRule(
+      getNewRule({
+        name: 'Successful rule',
+        rule_id: 'successful_rule',
+        index: ['test_index'],
+      })
+    );
+
+    createRule(
+      getNewRule({
+        name: 'Warning rule',
+        rule_id: 'warning_rule',
+        index: ['non_existent_index'],
+      })
+    );
+
+    createRule(
+      getNewRule({
+        name: 'Failed rule',
+        rule_id: 'failed_rule',
+        index: ['test_index'],
+        // Setting a crazy large "Additional look-back time" to force a failure
+        from: 'now-9007199254746990s',
+      })
+    );
+
+    waitForRulesToFinishExecution(['successful_rule', 'warning_rule', 'failed_rule']);
+
+    visitWithoutDateRange(SECURITY_DETECTIONS_RULES_URL);
+
+    waitForRulesTableToBeLoaded();
```

Comment:
I'm wondering if we need this in `beforeEach` -- were you thinking about expanding the rules you created with more properties in the future, to be able to test other filtering scenarios?

If we keep it as is, let's maybe also do some cleanup after the test:

```ts
    afterEach(() => {
      deleteAlertsAndRules();
      deleteIndex('test_index');
    });
```

Reply:
Thanks! I think it's a good idea to move non-common actions inside their respective `it` blocks.

Can please elaborate on why it would be beneficial to also clean up in `afterEach`? Wouldn't we do same actions twice then?

Reply:
> Can please elaborate on why it would be beneficial to also clean up in afterEach? Wouldn't we do same actions twice then?

@nikitaindik Yep, this is a good question. If we don't clean up, the rules we create in this test will continue to run in the next (and other) tests until a test cleans them up. This can have 2 downsides:

- The tests left running after this test will be generating load on the system when other tests will be starting/running. This could increase flakiness.
- If you ever need to debug those tests, you will see weird rule executions in the logs not related to the test you're debugging. This can be confusing.